Containing all gci-2014 commits

k-joseph opened the pull request:
This PR contains all commits previously merged into https://github.com/k-joseph/openmrs-core/tree/gci-2014

harsha89 commented on api/src/main/java/org/openmrs/Cohort.java:
```diff
@@ -27,13 +27,14 @@
 import org.simpleframework.xml.ElementList;
 import org.simpleframework.xml.Root;
+import org.springframework.expression.EvaluationContext;
 
```
Seems this is not used by the introduce changes!

k-joseph replied:
Actually shows up when i use the Ctrl + Shift + O option for resolving import issues

harsha89 commented on api/src/main/java/org/openmrs/Concept.java:
```diff
@@ -973,11 +974,10 @@ public void setShortName(ConceptName shortName) {
 			}
 			shortName.setConceptNameType(ConceptNameType.SHORT);
-			if (StringUtils.isNotBlank(shortName.getName())) {
+			if (StringUtils.isNotBlank(shortName.getName())
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/Concept.java:
```diff
@@ -973,11 +974,10 @@ public void setShortName(ConceptName shortName) {
-			if (StringUtils.isNotBlank(shortName.getName())) {
+			if (StringUtils.isNotBlank(shortName.getName())
+			        && (shortName.getConceptNameId() == null || !getNames().contains(shortName))) {
 				//add this name, if it is new or not among this concept's names
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
This is not gci, it is https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/Concept.java#L978 and am ok with it

harsha89 commented on api/src/main/java/org/openmrs/Concept.java:
```diff
@@ -1124,11 +1124,10 @@ public ConceptName getShortestName(Locale locale, Boolean exact) {
+				        && ((shortestNameForLocale == null) || (possibleName.getName().length() < shortestNameForLocale
+				                .getName().length()))) {
+					shortestNameForLocale = possibleName;
 				}
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
This looks to be just the same thing and my above response is ok, remembering that i created this Pull Request after checking out a new branch from master. Then pulled from upstream/gci-2014 and resolved conflicts which was the best way to confirm that every thing is updated and eliminate more conflicts

harsha89 commented on api/src/main/java/org/openmrs/ConceptAnswer.java:
```diff
@@ -251,4 +251,13 @@ public int compareTo(ConceptAnswer ca) {
 		}
 		return (getSortWeight() < ca.getSortWeight()) ? -1 : (getSortWeight() > ca.getSortWeight()) ? 1 : 0;
 	}
+	
```
Do we need this equals. Seems default equals does the same thing.

k-joseph replied:
Actually the modification just shows a space which is well and introduced by the openmrs formatter

harsha89 commented on api/src/main/java/org/openmrs/ConceptName.java:
```diff
@@ -526,10 +526,9 @@ public Boolean hasTag(String tagToFind) {
+		if (!StringUtils.isBlank(language) && this.locale != null && isPreferred()
+		        && this.locale.getLanguage().equals(language)) {
+			return true;
 		}
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/ConceptName.java:
```diff
@@ -545,10 +544,9 @@ public Boolean isPreferredInLanguage(String language) {
+		if (!StringUtils.isBlank(country) && this.locale != null && isPreferred()
+		        && this.locale.getCountry().equals(country)) {
+			return true;
 		}
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/ConceptName.java:
```diff
@@ -647,10 +645,8 @@ public Boolean isSynonym() {
-			}
+		if (!StringUtils.isBlank(language) && this.locale != null && isShort() && this.locale.getLanguage().equals(language)) {
+			return true;
 		}
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/ConceptName.java:
```diff
@@ -667,10 +663,8 @@ public Boolean isPreferredShortInLanguage(String language) {
-			}
+		if (!StringUtils.isBlank(country) && this.locale != null && isShort() && this.locale.getCountry().equals(country)) {
+			return true;
 		}
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/ConceptReferenceTerm.java:
```diff
@@ -169,16 +169,15 @@ public void setId(Integer id) {
-		if (conceptReferenceTermMap != null) {
+		if (conceptReferenceTermMap != null && conceptReferenceTermMap.getTermB() != null
+		        && !this.equals(conceptReferenceTermMap.getTermB())) {
 			//can't map a term to itself
```
Code make complex by aggregating this conditions. Should we have this or not?

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/ConceptSet.java:
```diff
@@ -206,4 +206,13 @@ public int compareTo(ConceptSet cs) {
 		}
 		return value;
 	}
+	
```
Default equlas does the same right?

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/FormField.java:
```diff
@@ -136,6 +136,15 @@ public int compareTo(FormField f) {
 		
 	}
 	
+	@Override
```
Same here!

harsha89 commented on api/src/main/java/org/openmrs/PatientIdentifier.java:
```diff
@@ -238,6 +238,15 @@ public int compareTo(PatientIdentifier other) {
 		return retValue;
 	}
 	
```
Same here for equlas

harsha89 commented on api/src/main/java/org/openmrs/PatientState.java:
```diff
@@ -206,4 +206,13 @@ public int compareTo(PatientState o) {
 		return result;
 	}
+	
+	@Override
```
same here for equals

k-joseph replied:
See  my above responses :+1:

harsha89 commented on api/src/main/java/org/openmrs/PersonAttribute.java:
```diff
@@ -322,6 +322,15 @@ public int compareTo(PersonAttribute other) {
 		return OpenmrsUtil.compareWithNullAsGreatest(getPersonAttributeId(), other.getPersonAttributeId());
 	}
 	
+	@Override
```
Same for equals

harsha89 commented on api/src/main/java/org/openmrs/PersonAttributeType.java:
```diff
@@ -183,4 +183,13 @@ public void setId(Integer id) {
 	public int compareTo(PersonAttributeType other) {
 		return OpenmrsUtil.compareWithNullAsGreatest(getPersonAttributeTypeId(), other.getPersonAttributeTypeId());
 	}
+	
```
Same

harsha89 commented on api/src/main/java/org/openmrs/activelist/Problem.java:
```diff
@@ -120,4 +120,13 @@ public int compareTo(Problem item) {
 		
 		return mySW.compareTo(theirSW);
 	}
+	
```
Same here on equals

harsha89 commented on api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java:
```diff
@@ -54,10 +54,9 @@ public boolean match(MetadataReader metadataReader, MetadataReaderFactory metada
+		if (StringUtils.isNotBlank((String) openmrsVersion)
+		        && !ModuleUtil.matchRequiredVersions(OpenmrsConstants.OPENMRS_VERSION_SHORT, (String) openmrsVersion)) {
+			return false;
 		}
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java:
```diff
@@ -69,11 +68,10 @@ public boolean matchOpenmrsProfileAttributes(Map<String, Object> openmrsProfile)
+				        && ModuleUtil.matchRequiredVersions(module.getVersion(), moduleVersion)) {
+					moduleMatched = true;
+					break;
 				}
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/api/context/UserContext.java:
```diff
@@ -309,13 +309,12 @@ public Locale getLocale() {
-		if (isAuthenticated()) {
+		if (isAuthenticated()
+		        && (getAuthenticatedUser().hasPrivilege(privilege) || getAuthenticatedRole().hasPrivilege(privilege))) {
 			
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/api/db/hibernate/AuditableInterceptor.java:
```diff
@@ -165,13 +165,9 @@ private boolean changePropertyValue(Object[] currentState, String[] propertyName
+		if (index >= 0 && (currentState[index] == null || !setNullOnly) && !value.equals(currentState[index])) {
+			currentState[index] = value;
+			return true;
 		}
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/api/handler/BaseUnretireHandler.java:
```diff
@@ -54,14 +54,13 @@
-		if (retireableObject.isRetired()) {
+		if (retireableObject.isRetired()
+		        && (origParentRetiredDate == null || origParentRetiredDate.equals(retireableObject.getDateRetired()))) {
 			// only act on retired objects that match the same date retired as the parent
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/api/handler/BaseUnvoidHandler.java:
```diff
@@ -54,15 +54,14 @@
-		if (voidableObject.isVoided()) {
+		if (voidableObject.isVoided()
+		        && (origParentVoidedDate == null || origParentVoidedDate.equals(voidableObject.getDateVoided()))) {
 			
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/api/handler/OpenmrsObjectSaveHandler.java:
```diff
@@ -111,11 +111,9 @@ public void handle(OpenmrsObject openmrsObject, User creator, Date dateCreated,
 				
-				if ("".equals(value)) {
+				if ("".equals(value) && !(openmrsObject instanceof Voidable && ((Voidable) openmrsObject).isVoided())) {
 					//Set to null only if object is not already voided
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/api/impl/DatatypeServiceImpl.java:
```diff
@@ -60,7 +60,7 @@ public void setDao(DatatypeDAO dao) {
 	 */
```
Do we need really need these sysnchornized blocks?

harsha89 commented on api/src/main/java/org/openmrs/attribute/BaseAttribute.java:
```diff
@@ -149,4 +149,12 @@ public int compareTo(Attribute other) {
 		return retValue;
 	}
 	
+	@Override
```
Do we need this equals?

harsha89 commented on api/src/main/java/org/openmrs/layout/LayoutSupport.java:
```diff
@@ -59,13 +59,12 @@ public T getLayoutTemplateByName(String templateName) {
-					        || templateName.equalsIgnoreCase(at.getCountry())) {
-						ret = at;
-						log.debug("Found Layout Template named " + at.getDisplayName());
-					}
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/layout/LayoutSupport.java:
```diff
@@ -81,11 +80,9 @@ public T getLayoutTemplateByCodeName(String templateName) {
-					if (templateName.equalsIgnoreCase(at.getCodeName())) {
-						ret = at;
-						log.debug("Found Layout Template named " + at.getDisplayName());
-					}
```
Code make complex by aggregating this conditions. Should we have this or not?

harsha89 commented on api/src/main/java/org/openmrs/obs/handler/TextHandler.java:
```diff
@@ -132,6 +132,7 @@ public Obs saveObs(Obs obs) throws APIException {
 						}
 						fout.write(character);
 					}
+					tempRd.close();
```
better do it in a finally block

harsha89 commented on api/src/main/java/org/openmrs/util/UpgradeUtil.java:
```diff
@@ -43,15 +43,18 @@ public static Integer getConceptIdForUnits(String units) {
 					conceptId = prop.getValue().toString();
 					
 					if (conceptId != null) {
+						fis.close();
```
Need to close streams in a finally blocks

harsha89 commented on api/src/main/java/org/openmrs/util/UpgradeUtil.java:
```diff
@@ -43,15 +43,18 @@ public static Integer getConceptIdForUnits(String units) {
+						fis.close();
 						return Integer.valueOf(conceptId);
 					} else {
+						fis.close();
```
Need to close streams in a finally blocks

harsha89 commented on web/src/main/java/org/openmrs/web/controller/encounter/EncounterDisplayController.java:
```diff
@@ -390,6 +390,14 @@ public int compareTo(FieldHolder other) {
 			return temp;
 		}
 		
```
Equals needed?

harsha89 commented on web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java:
```diff
@@ -949,7 +949,9 @@ private boolean verifyConnection(String connectionUsername, String connectionPas
 			//Set Database Driver using driver String
 			Class.forName(loadedDriverString).newInstance();
-			DriverManager.getConnection(databaseConnectionFinalUrl, connectionUsername, connectionPassword);
+			Connection tempConnection = DriverManager.getConnection(databaseConnectionFinalUrl, connectionUsername,
```
Need to close conncetion in finaly block

harsha89 commented on web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java:
```diff
@@ -1132,7 +1134,10 @@ private int executeStatement(boolean silent, String user, String pw, String sql,
-			return statement.executeUpdate(replacedSql);
+			
+			int updateDelta = statement.executeUpdate(replacedSql);
+			statement.close();
```
close connections in a finally block

harsha89 commented on web/src/main/java/org/openmrs/web/filter/update/UpdateFilter.java:
```diff
@@ -350,9 +350,13 @@ protected boolean authenticateAsSuperUser(String usernameOrSystemId, String pass
 				}
 			}
+			// Close statement
+			statement.close();
```
Stmt should close in finally blocls

harsha89 commented on web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java:
```diff
@@ -71,6 +71,9 @@ public static String restoreLocale(String username) {
 					}
+					
+					//close statement
+					statement.close();
```
same as above

harsha89 commented on web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java:
```diff
@@ -129,14 +132,12 @@ public static String readSystemDefaultLocale(Connection connection) {
-					}
+			if (needToCloseConection && connection != null) {
+				try {
+					connection.close();
```
same as abve

harsha89 commented on web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java:
```diff
@@ -180,6 +181,8 @@ public static boolean storeLocale(String locale) {
 					log.warn("Unable to set system default locale property.");
 				}
+				// close statement to release resources
+				statement.close();
```
same as abve

k-joseph commented:
@dkayiwa, my last commit here at: https://github.com/k-joseph/openmrs-core/commit/91c6be799f090fa3b1fe1a9c325cdb4b68da1905 addresses your reviews, we can later handle the log and deprecated issues
